### PR TITLE
Three breaking changes

### LINKED ISSUE #36720: [Breaking change]: System.Configuration.ConfigurationManager no longer references System.Security.Permissions
### Description

The `System.Configuration.ConfigurationManager` package no longer references the `System.Security.Permissions` package.

### Version

.NET 8 Preview 3

### Previous behavior

The `System.Configuration.ConfigurationManager` referenced the `System.Security.Permissions` package.

### New behavior

The `System.Configuration.ConfigurationManager` does not reference the `System.Security.Permissions` package.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load or execute, and if so, require recompilation.
- [x] **Source incompatible**: When recompiled using the new SDK or component or to target the new runtime, existing source code may require source changes to compile successfully.
- [ ] **Behavioral change**: Existing binaries may behave differently at run time.

### Reason for change

This avoids a dependency on `System.Drawing.Common` when `System.Configuration.ConfigurationManager` is referenced which is primarily an issue for non-Windows operating systems.

The dependency to `System.Drawing.Common` was caused by the following package dependencies:
System.Configuration.ConfigurationManager ->
System.Security.Permissions ->
System.Windows.Extensions ->
System.Drawing.Common.

### Recommended action

If the `System.Configuration.ConfigurationManager` package is referenced and a dependency is needed to `System.Security.Permissions` or any of its dependencies, which may include `System.Windows.Extensions`, `System.Security.AccessControl`, or `System.Drawing.Common` then ensure those packages are referenced directly or indirectly.

### Feature area

Other (please put exact area in description textbox)

### Affected APIs

_No response_

### LINKED ISSUE #36724: [Breaking change]: System.DirectoryServices no longer references System.Security.Permissions
### Description

The `System.DirectoryServices` package no longer references the `System.Security.Permissions` package.

### Version

.NET 8 Preview 3

### Previous behavior

The `System.DirectoryServices` package referenced the `System.Security.Permissions` package.

### New behavior

The `System.DirectoryServices` package does not reference the `System.Security.Permissions` package.

### Type of breaking change

- [ ] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load or execute, and if so, require recompilation.
- [x] **Source incompatible**: When recompiled using the new SDK or component or to target the new runtime, existing source code may require source changes to compile successfully.
- [ ] **Behavioral change**: Existing binaries may behave differently at run time.

### Reason for change

This avoids a dependency on `System.Drawing.Common` when `System.DirectoryServices` is referenced which is primarily an issue for non-Windows operating systems.

The dependency to `System.Drawing.Common` was caused by the following package dependencies:
System.DirectoryServices ->
System.Security.Permissions ->
System.Windows.Extensions ->
System.Drawing.Common.

### Recommended action

If the `System.DirectoryServices` package is referenced and a dependency is needed to `System.Security.Permissions` or any of its dependencies, which may include `System.Windows.Extensions` or `System.Drawing.Common` then ensure those packages are referenced directly or indirectly.

### Feature area

Other (please put exact area in description textbox)

### Affected APIs

_No response_

### LINKED ISSUE #36725: [Breaking change]: ConstructorBuilder and MethodBuilder now generate method parameters that when reflected on have ParameterInfo.HasDefaultValue=false
### Description

`System.Reflection.Emit.ConstructorBuilder` and `System.Reflection.Emit.MethodBuilder` generate IL for the parameters where the `HasDefaultValue` of the parameters returns `true` which is not correct.

### Version

.NET 8 Preview 5

### Previous behavior

`System.Reflection.Emit.ConstructorBuilder` and `MethodBuilder` generate IL for the parameters where the `HasDefaultValue` of the parameters is `true`.

### New behavior

`System.Reflection.Emit.ConstructorBuilder` and `MethodBuilder` generate IL for the parameters where the `HasDefaultValue` of the parameters is `false`, as expected.

### Type of breaking change

- [X] **Binary incompatible**: Existing binaries may encounter a breaking change in behavior, such as failure to load or execute, and if so, require recompilation.
- [ ] **Source incompatible**: When recompiled using the new SDK or component or to target the new runtime, existing source code may require source changes to compile successfully.
- [ ] **Behavioral change**: Existing binaries may behave differently at run time.

### Reason for change

The previous behavior was incorrect, as no default parameter values were specified when the method or constructor was defined.

### Recommended action

If `System.Reflection.Emit.TypeBuilder.DefineConstructor()` or `System.Reflection.Emit.TypeBuilder.DefineMethod()` are used, ensure that consumers of the generated types' methods do not rely on the `ParameterInfo.HasDefaultValue` property being `true`.

### Feature area

Core .NET libraries

### Affected APIs

`System.Reflection.ParameterInfo.HasDefaultValue`

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #36720 
Fixes #36724 
Fixes #36725 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/8.0.md](https://github.com/dotnet/docs/blob/20a2266c04a6e96533b263cd1bd446379854f3d3/docs/core/compatibility/8.0.md) | [Breaking changes in .NET 8](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/8.0?branch=pr-en-us-36787) |
| [docs/core/compatibility/core-libraries/8.0/parameterinfo-hasdefaultvalue.md](https://github.com/dotnet/docs/blob/20a2266c04a6e96533b263cd1bd446379854f3d3/docs/core/compatibility/core-libraries/8.0/parameterinfo-hasdefaultvalue.md) | [docs/core/compatibility/core-libraries/8.0/parameterinfo-hasdefaultvalue](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/8.0/parameterinfo-hasdefaultvalue?branch=pr-en-us-36787) |
| [docs/core/compatibility/extensions/8.0/configurationmanager-package.md](https://github.com/dotnet/docs/blob/20a2266c04a6e96533b263cd1bd446379854f3d3/docs/core/compatibility/extensions/8.0/configurationmanager-package.md) | [docs/core/compatibility/extensions/8.0/configurationmanager-package](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/extensions/8.0/configurationmanager-package?branch=pr-en-us-36787) |
| [docs/core/compatibility/extensions/8.0/directoryservices-package.md](https://github.com/dotnet/docs/blob/20a2266c04a6e96533b263cd1bd446379854f3d3/docs/core/compatibility/extensions/8.0/directoryservices-package.md) | [docs/core/compatibility/extensions/8.0/directoryservices-package](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/extensions/8.0/directoryservices-package?branch=pr-en-us-36787) |

### CODE REVIEW

File: docs/core/compatibility/extensions/8.0/configurationmanager-package.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+This change avoids a dependency on `System.Drawing.Common` when `System.Configuration.ConfigurationManager` is referenced, which is primarily an issue for non-Windows operating systems.
+
+The dependency on `System.Drawing.Common` was caused by the following package dependencies:
+
+```txt
+System.Configuration.ConfigurationManager ->
+System.Security.Permissions ->
+System.Windows.Extensions ->
+System.Drawing.Common
```
Comment (IEvangelist) [inline docs/core/compatibility/extensions/8.0/configurationmanager-package.md:36]:
```suggestion
System.Configuration.ConfigurationManager
└──System.Security.Permissions
    └──System.Windows.Extensions
        └──System.Drawing.Common
```

File: docs/core/compatibility/extensions/8.0/directoryservices-package.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+.NET 8 Preview 3
+
+## Previous behavior
+
+The `System.DirectoryServices` package referenced the `System.Security.Permissions` package.
+
+## New behavior
+
+Starting in .NET 8, the `System.DirectoryServices` package *does not reference* the `System.Security.Permissions` package.
```
Comment (IEvangelist) [inline docs/core/compatibility/extensions/8.0/directoryservices-package.md:20]:
```suggestion
Starting in .NET 8, the `System.DirectoryServices` package *doesn't reference* the `System.Security.Permissions` package.
```

File: docs/core/compatibility/extensions/8.0/directoryservices-package.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+This change avoids a dependency on `System.Drawing.Common` when `System.DirectoryServices` is referenced, which is primarily an issue for non-Windows operating systems.
+
+The dependency on `System.Drawing.Common` was caused by the following package dependencies:
+
+```txt
+System.DirectoryServices ->
+System.Security.Permissions ->
+System.Windows.Extensions ->
+System.Drawing.Common
```
Comment (IEvangelist) [inline docs/core/compatibility/extensions/8.0/directoryservices-package.md:36]:
```suggestion
System.DirectoryServices
 └──System.Security.Permissions
      └──System.Windows.Extensions
           └──System.Drawing.Common
```

**Review by IEvangelist — APPROVED** [review]
A few nits/suggestions

File: docs/core/compatibility/extensions/8.0/configurationmanager-package.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+.NET 8 Preview 3
+
+## Previous behavior
+
+The `System.Configuration.ConfigurationManager` package referenced the `System.Security.Permissions` package.
+
+## New behavior
+
+Starting in .NET 8, the `System.Configuration.ConfigurationManager` package *does not reference* the `System.Security.Permissions` package.
```
[the remark this answers is not in the document]
Comment (gewarren) [reply docs/core/compatibility/extensions/8.0/configurationmanager-package.md:20]:
```suggestion
Starting in .NET 8, the `System.Configuration.ConfigurationManager` package *does not* reference the `System.Security.Permissions` package.
```

File: docs/core/compatibility/extensions/8.0/directoryservices-package.md (hunk 2 above)
Comment (gewarren) [reply docs/core/compatibility/extensions/8.0/directoryservices-package.md:20]:
```suggestion
Starting in .NET 8, the `System.DirectoryServices` package *does not* reference the `System.Security.Permissions` package.
```

**Pull request merged**